odoo/addons/base/models/res_users.py

```diff
@@ -555,13 +556,22 @@ def _check_action_id(self):
         action_open_website = self.env.ref('base.action_open_website', raise_if_not_found=False)
         if action_open_website and any(user.action_id.id == action_open_website.id for user in self):
             raise ValidationError(_('The "App Switcher" action cannot be selected as home action.'))
-        # Prevent using reload actions.
+
         # We use sudo() because  "Access rights" admins can't read action models
         for user in self.sudo():
             if user.action_id.type == "ir.actions.client":
+                # Prevent using reload actions.
                 action = self.env["ir.actions.client"].browse(user.action_id.id)  # magic
                 if action.tag == "reload":
                     raise ValidationError(_('The "%s" action cannot be selected as home action.', action.name))
+            elif user.action_id.type == "ir.actions.act_window":
+                # Restrict actions that include 'active_id' in their context.
+                action = self.env["ir.actions.act_window"].browse(user.action_id.id)  # magic
+                if not action.context:
+                    continue
+                found_restricted_key = re.search(r":\s*active_id\s*[,}]", action.context)
+                if found_restricted_key:
+                    raise ValidationError(_('The "%s" action cannot be selected as home action.', action.name))
```

Comment:
Should we need to explain why it is not selectable ?

Reply:
I will update the commit message to explain why it shoudnt be selected.